src/buffer.rs

```diff
@@ -239,3 +313,101 @@ pub enum IQMatrix {
     /// Abstraction over `VAIQMatrixBufferHEVC`
     HEVC(hevc::IQMatrixBufferHEVC),
 }
+
+pub enum EncSequenceParameter {
+    H264(h264::EncSequenceParameterBufferH264),
+}
+
+pub enum EncPictureParameter {
+    H264(h264::EncPictureParameterBufferH264),
+}
+
+pub enum EncSliceParameter {
+    H264(h264::EncSliceParameterBufferH264),
+}
+
+pub enum EncMacroblockParameterBuffer {
+    H264(h264::EncMacroblockParameterBufferH264),
+}
+
+pub enum EncCodedBuffer {
+    UndefinedStorage(usize),
+    LocalStorage(Vec<u8>), // This one does not work :C
```

Comment:
I do not think we need this.

```
 * Basically there are two ways to get buffer data to the server side. One is
 * to call vaCreateBuffer() with a non-null "data", which results the data being
 * copied to the data store on the server side.  A different method that
 * eliminates this copy is to pass null as "data" when calling vaCreateBuffer(),
 * and then use vaMapBuffer() to map the data store from the server side to the
 * client address space for access.
 ```
 
 We're trying to get data *from* the server, so the `LocalStorage` variant looks unnecessary.